### ISSUE TITLE
sensu-plugin is "removed" every puppet run

### ISSUE THREAD

**Issue by robinbowes** [sensu/sensu-puppet#298]
puppet attempts to remove the sensu-plugin package every run, producing this output:
```
Info: Applying configuration version '1421975032'
Notice: /Stage[main]/Sensu::Package/Package[sensu-plugin]/ensure: removed
Info: Class[Sensu::Package]: Scheduling refresh of Service[sensu-client]
Info: Class[Sensu::Package]: Scheduling refresh of Service[sensu-api]
Info: Class[Sensu::Package]: Scheduling refresh of Service[sensu-server]
Notice: /Stage[main]/Sensu::Client::Service/Service[sensu-client]: Triggered 'refresh' from 1 events
Notice: /Stage[main]/Sensu::Api::Service/Service[sensu-api]: Triggered 'refresh' from 1 events
Notice: /Stage[main]/Sensu::Server::Service/Service[sensu-server]: Triggered 'refresh' from 1 events
Notice: Finished catalog run in 7.95 seconds
```
This is puppet 3.7.3 on CentOS 7.

**Comment by robinbowes** [comment]
This seems to be because sensu_plugin_provider defaults to `undef`, which means $plugin_provider gets set to 'sensu_gem', which causes the package resource to always run to remove the sensu-plugin package using the sensu_gem provider.

A workaround for the issue is to explicitly set the sensu_plugin_provider to the default for the platform (I'm using CentOS 7, so "yum"). That prevents the problem.

A better fix would be to re-work the logic around setting the $plugin_provider variable.

**Comment by nhinds** [comment]
My workaround for this is to set `sensu::sensu_plugin_version` to `present` which ensures it's in the correct state.

This is only an issue if you have `sensu::use_embedded_ruby` set to `true` (or if you have it set to `false` and you want `sensu-plugin` as one of your system gems, I guess, but I haven't tested that scenario)

**Comment by YHSC** [comment]
@nhinds thanks for the hint wrt embedded ruby, solved my issue. would be good to have this documented.

**Comment by queeno** [comment]
This issue has been fixed in 78236fd

I believe this issue has been now resolved.

**Comment by devopsprosiva** [comment]
I still see the issue. I'm using Puppet 4 and CentOS 7.

**Comment by devshorts** [comment]
same

**Comment by jlambert121** [comment]
@devopsprosiva and @devshorts are you still seeing the issue running from master and restarting the puppet master after upgrading the sensu module?  (It is important to restart the puppet master so it reloads the types and providers)

**Comment by jlambert121** [comment]
If this is still an issue, we'll hopefully get it resolved as part of fixing #328.
